scripts/artifacts/recentactivity.py

```diff
@@ -0,0 +1,216 @@
+                    component = (child.attrib.get('component'))
+                    icon_image_path = (root.attrib.get('task_description_icon_filename'))
+                    
+                    #Snapshot section picture
+                    snapshot = task_id + '.jpg'
+                    #print(snapshot)
+                    
+                    #check for image in directories
+                    check1 = os.path.join(script_dir, 'snapshots', snapshot)
+                    isit1 = os.path.isfile(check1)
+                    if isit1:
+                        #copy snaphot image to report folder
+                        shutil.copy2(check1, report_folder)
+                        #snap = r'./snapshots/' + snapshot
+                        snap = snapshot
+                    else:
+                        snap = 'NO IMAGE'
+                    #Recent_images section
+                    if icon_image_path is not None:
+                        recent_image = os.path.basename(icon_image_path)
+                        check2 = os.path.join(script_dir, 'recent_images', recent_image)
+                        isit2 = os.path.isfile(check2)
+                        if isit2:
+                            shutil.copy2(check2, report_folder)
+                            #recimg = r'./recent_images/' + recent_image
+                            recimg = recent_image
+                        else:
+                            recimg = 'NO IMAGE'
+                    else:
+                        #check for other files not in the XML - all types
+                        check3 = glob.glob(os.path.join(script_dir, 'recent_images', task_id, '*.*'))
+                        if check3:
+                            check3 = check3[0]
+                            isit3 = os.path.isfile(check3)
+                        else:
+                            isit3 = 0
+                        
+                        if isit3:
+                            shutil.copy2(check3, report_folder)
+                            recimg = os.path.basename(check3)
+                        else:
+                            recimg = 'NO IMAGE'
+                    #else:
+                    #    recimg = 'NO IMAGE'
+                    #insert all items in database
+                    cursor = db.cursor()
+                    datainsert = (task_id, effective_uid, affinity, real_activity, first_active_time, last_active_time, last_time_moved, calling_package, user_id, action, component, snap, recimg, fullat1, fullat2,)
+                    cursor.execute('INSERT INTO recent (task_id, effective_uid, affinity, real_activity, first_active_time, last_active_time, last_time_moved, calling_package, user_id, action, component, snap, recimg, fullat1, fullat2)  VALUES(?,?,?,?,?,?,?,?,?,?,?,?,?,?,?)', datainsert)
+                    db.commit()
+                    
+    #Create html file for report
+    f1= open(os.path.join(report_folder, 'Recent Activity.html'), 'w+')
```

Comment:
I personally prefer using a `with` context manager.

This is ok because the process will eventually exit, so one can trust the OS to clean up the file handles in most cases but using a context manager is the easiest way to not have to deal with that in case execution gets interrupted before you hit `f1.close()`.

Reply:
Yeah that is my code. This is what I get for trying to put put together old code with new. You are absolutely correct.

Reply:
Cool cool, we'll get around to cleaning it all up eventually.